drivers/wifi/esp_at/esp.c

```diff
@@ -276,7 +277,18 @@ MODEM_CMD_DEFINE(on_cmd_cwlap)
 	memcpy(res.ssid, argv[1], i);
 	res.ssid_length = i;
 	res.rssi = strtol(argv[2], NULL, 10);
-	res.channel = strtol(argv[3], NULL, 10);
+
+	if (IS_ENABLED(CONFIG_WIFI_ESP_AT_SCAN_MAC_ADDRESS)) {
+		argv[3] = str_unquote(argv[3]);
+		res.mac_length = WIFI_MAC_ADDR_LEN;
```

Comment:
There is no `res.mac_length` at the time you add this commit. You should reorder commits to make sure Zephyr codebase will be bisectable.

Reply:
Reordered the commits.